tests/test_normalized.cpp

```diff
@@ -0,0 +1,221 @@
+ *   it under the terms of the GNU General Public License as published by    *
+ *   the Free Software Foundation; either version 2 of the License, or       *
+ *   (at your option) any later version.                                     *
+ *                                                                           *
+ *   This program is distributed in the hope that it will be useful,         *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of          *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the           *
+ *   GNU General Public License for more details.                            *
+ *                                                                           *
+ *   You should have received a copy of the GNU General Public License       *
+ *   along with this program; if not, write to the                           *
+ *   Free Software Foundation, Inc.,                                         *
+ *   59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.               *
+ *****************************************************************************/
+
+// Test code for the normalized meta-problem
+
+#include <iostream>
+#include <algorithm>
+#include <cmath>
+#include <vector>
+#include <cassert>
+#include "../src/pagmo.h"
+#include "../src/rng.h"
+#include <boost/random/uniform_real.hpp>
+
+using namespace pagmo;
+
+const double EPS = 10e-9;
+
+#define PRINT_VEC(x) do {                             \
+	std::cout << "[ " #x " ] = ";                     \
+	for(unsigned int iii=0; iii<(x).size(); iii++) {  \
+		std::cout<<(x)[iii]<<" ";                     \
+	}                                                 \
+	std::cout<<std::endl; } while(0);
+
+bool is_eq(fitness_vector f1, fitness_vector f2, double eps)
+{
+	if (f1.size() != f2.size()) {
+		return false;
+	}
+	for (size_t i = 0; i < f1.size(); i++) {
+		if (fabs(f1[i] - f2[i]) > eps) {
+			return false;
+		}
+	}
+	return true;
+}
+
+// Test following invariant:
+// normalized(prob).objfun(dv) is equal to
+//   prob.objfun(normalized(prob).denormalize(dv))
+int test_normalized_invariant(
+	const std::vector<problem::base_ptr> &probs)
+{
+	rng_double drng	= rng_generator::get<rng_double>();
+
+	std::cout << "Start batch testing of normalization invariant" << std::endl;
+
+	for (size_t i = 0; i < probs.size(); ++i) {
+		problem::base_ptr cur_prob = probs[i]->clone();
+		size_t dim = cur_prob->get_dimension();
+		decision_vector p_test(dim);
+
+		std::cout<< std::setw(40) << cur_prob->get_name();
+
+		// Normalize problem.
+		pagmo::problem::normalized norm(*(cur_prob));
+
+		if (dim != norm.get_lb().size()) {
+			std::cout << " Unexpected bounds vector size" << std::endl;
+			std::cout << " Expected " << dim << " got " << norm.get_lb().size();
+			return 1;
+		}
+
+		// Test that we have indeed normalized: lb[i] = -1 and ub[i] = 1.
+		for (size_t k = 0; k < dim; ++k) {
+			if (fabs(norm.get_lb()[k] - (-0.1) > EPS) ||
```

Comment:
I think you got the brackets here wrong, e.g. ``fabs`` encapsulates the `` > EPS``, i think you meant:

```c++
if (fabs(norm.get_lb()[k] - (-1.0)) > EPS || fabs(norm.get_ub()[k] - (1.0)) > EPS)
```

but again, I think we should reuse some "common" debugging functions, such as:

```c++
bool is_eq(double d1, double d2, double eps);
bool is_eq_v(fitness_vector v1, fitness_vector v2, double eps);  // applies is_eq to each element
```